### PR TITLE
[MLDB-1380] returning a specific error message when the cpu does not …

### PR DESCRIPTION
**Pull request by guyd**
…support sse 4.2

### CODE REVIEW

File: server/test_arch.cc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include "mldb/arch/simd.h"
```
Comment (guyd) [inline server/test_arch.cc]: This file will not be committed.

File: server/mldb_server.cc [hunk 2]
```diff
@@ -187,26 +188,49 @@ initRoutes()
...
-                                         true),
-                  RestParamDefault<bool>("rowNames",
-                                         "Do we include row names in output",
-                                         true),
-                  RestParamDefault<bool>("rowHashes",
-                                         "Do we include row hashes in output",
-                                         false));
+
+   // MLDB-1380 - make sure that the CPU support the minimal instruction sets
+    if (ML::has_sse42()) {
```
Comment (jeremybarnes) [inline server/mldb_server.cc]: Consider splitting this out into a separate function like supportsSystemRequirements()

Comment (guyd) [inline server/mldb_server.cc]: You mean the if condition or the whole if statement?

Comment (jeremybarnes) [inline server/mldb_server.cc]: the condition.  +1 once that's done

**Pull request merged**